test/library/etcd.go

```diff
@@ -111,55 +129,131 @@ func NewEtcdKV(kubeClient kubernetes.Interface) (clientv3.KV, func(), error) {
 	return etcdClient3.KV, done, nil
 }
 
-func AssertEtcdSecretNotEncrypted(t *testing.T, kv clientv3.KV, namespace, name string) {
-	t.Helper()
-	secret := GetEtcdSecretMust(t, kv, namespace, name)
+func CheckEncryptionState(expectedMode string) func([]byte) error {
```

Comment:
would you consider writing a test or two that compare the content and not only the prefix? this would protect in cases where the prefix matched but the content isn't encrypted.

Reply:
I will explore this in a follow-up.